File: packages/sdk-ruby/lib/bidstack_crm/resources/notifications.rb
require_relative "base"

module BidstackCrm
  module Resources
    class Notifications < Base
      def list(page: 1, per_page: 50, unread_only: false)
        params = { page: page, perPage: per_page }
        params[:unreadOnly] = true if unread_only
        get("/notifications", params: params)
      end

      def mark_read(notification_id) = patch("/notifications/#{notification_id}", body: { read: true })
      def mark_all_read = post("/notifications/mark-all-read")
    end
  end
end

File: packages/sdk-python/bidstack_crm/resources/notifications.py
"""Notifications resource."""
from __future__ import annotations
from typing import Any
from ._base import BaseResource


class NotificationsResource(BaseResource):
    def list(self, *, unread_only: bool = False, page: int = 1, per_page: int = 50) -> dict[str, Any]:
        params: dict[str, Any] = {"page": page, "perPage": per_page}
        if unread_only:
            params["unreadOnly"] = "true"
        return self._get("/notifications", params=params)

    def mark_read(self, notification_id: str) -> dict[str, Any]:
        return self._patch(f"/notifications/{notification_id}", json={"read": True})

    def mark_all_read(self) -> dict[str, Any]:
        return self._post("/notifications/mark-all-read")
